plugins/obs-filters/nvidia-audiofx-filter.c

```diff
@@ -0,0 +1,924 @@
+	case 3:
+		return SPEAKERS_2POINT1;
+	case 4:
+		return SPEAKERS_4POINT0;
+	case 5:
+		return SPEAKERS_4POINT1;
+	case 6:
+		return SPEAKERS_5POINT1;
+	case 8:
+		return SPEAKERS_7POINT1;
+	default:
+		return SPEAKERS_UNKNOWN;
+	}
+}
+
+static void set_nv_model(void *data, const char *method)
+{
+	struct nvidia_audio_data *ng = data;
+	const char *file;
+	if (strcmp(NVAFX_EFFECT_DEREVERB, method) == 0)
+		file = NVAFX_EFFECT_DEREVERB_MODEL;
+	else if (strcmp(NVAFX_EFFECT_DEREVERB_DENOISER, method) == 0)
+		file = NVAFX_EFFECT_DEREVERB_DENOISER_MODEL;
+	else
+		file = NVAFX_EFFECT_DENOISER_MODEL;
+	size_t size = strlen(ng->sdk_path) + strlen(file) + 1;
+	char *buffer = (char *)bmalloc(size);
+
+	strcpy(buffer, ng->sdk_path);
+	strcat(buffer, file);
+	ng->model = buffer;
+}
+
+static void nvidia_audio_update(void *data, obs_data_t *s)
+{
+	struct nvidia_audio_data *ng = data;
+
+	if (!ng->use_nvafx)
+		return;
+	const char *method = obs_data_get_string(s, S_METHOD);
+	ng->latency = 1000000000LL / (1000 / BUFFER_SIZE_MSEC);
+
+	float intensity = (float)obs_data_get_double(s, S_NVAFX_INTENSITY);
+	/*-------------------------------------------------------------------*/
+	/* STAGE 1 : the following is run only when the filter is created. */
+
+	/* If the DLL hasn't been loaded & isn't loading, start the side loading */
+	if (!ng->nvafx_initialized && !ng->nvafx_loading) {
+		ng->intensity_ratio = intensity;
+		ng->nvafx_loading = true;
+		pthread_create(&ng->nvafx_thread, NULL, nvidia_audio_initialize,
+			       ng);
+	}
+
+	/*-------------------------------------------------------------------*/
+	/* STAGE 2 : this is executed only after the FX has been initialized */
+	if (ng->nvafx_initialized) {
+		/* updating the intensity of the FX */
+		if (intensity != ng->intensity_ratio &&
+		    (strcmp(ng->fx, method) == 0)) {
+			NvAFX_Status err;
+			ng->intensity_ratio = intensity;
+			pthread_mutex_lock(&ng->nvafx_mutex);
+			for (size_t i = 0; i < ng->channels; i++) {
+				err = NvAFX_SetFloat(
+					ng->handle[i],
+					NVAFX_PARAM_INTENSITY_RATIO,
+					ng->intensity_ratio);
+				if (err != NVAFX_STATUS_SUCCESS) {
+					do_log(LOG_ERROR,
+					       "NvAFX_SetFloat(Intensity Ratio: %f) failed, error %i",
+					       ng->intensity_ratio, err);
+					nvidia_audio_destroy(ng);
+				}
+			}
+			pthread_mutex_unlock(&ng->nvafx_mutex);
+		}
+		/* swapping to a new FX requires a reinitialization */
+		if ((strcmp(ng->fx, method) != 0)) {
+			pthread_mutex_lock(&ng->nvafx_mutex);
+			bfree((void *)ng->fx);
+			ng->fx = bstrdup(method);
+			ng->intensity_ratio = intensity;
+			set_nv_model(ng, method);
+			os_atomic_set_bool(&ng->reinit_done, false);
+			for (int i = 0; i < (int)ng->channels; i++) {
+				/* Destroy previous FX */
+				if (NvAFX_DestroyEffect(ng->handle[i]) !=
+				    NVAFX_STATUS_SUCCESS) {
+					do_log(LOG_ERROR,
+					       "FX failed to be destroyed.");
+					nvidia_audio_destroy(ng);
+				} else
+					ng->handle[i] = NULL;
```

Comment:
If one branch of an `if` statement has curly braces, they must all have curly braces.

Reply:
fixed